src/experiment.ts

```diff
@@ -0,0 +1,127 @@
+import { Identify } from 'amplitude-js';
+import { Client } from './client';
+
+const LOCAL_STORAGE_EXPERIMENTS_PREFIX = '__analytics_experiments_';
+
+/** Experiment interface allows getting a variation for a specified user. */
+export interface Experiment<Variation extends string> {
+	name: string;
+
+	engage(userId: string): Variation;
+}
+
+interface VariationData<Variation> {
+	variation: Variation;
+	fraction: number;
```

Comment:
Let's rename this to something more descriptive, such as `targetPercentage`, as `fraction` doesn't imply the value is between 0 and 100, it can be anything.